webapp/lib/app/nook/view.dart

```diff
@@ -1135,52 +1135,47 @@ class ConversationFilter {
-
-      newContainerTitle.onClick.listen((event) {
-          newContainer.classes.toggle('hidden');
-          newContainerTitle.classes.toggle('folded');
-      });
-      // Start off folded
-      newContainer.classes.toggle('hidden', true);
-      newContainerTitle.classes.toggle('folded', true);
-
-      return newContainer;
-    }).append(tag.renderElement);
-    bool wasHidden = _tagGroupsContainers[category].classes.remove('hidden'); // briefly override any display settings to make sure we can compute getBoundingClientRect()
-    List<num> widths = _tagGroupsContainers[category].querySelectorAll('.tag__text').toList().map((e) => e.getBoundingClientRect().width).toList();
-    _tagGroupsContainers[category].classes.toggle('hidden', wasHidden); // clear inline display settings
-    num avgGridWidth = widths.fold(0, (previousValue, width) => previousValue + width);
-    avgGridWidth = avgGridWidth / widths.length;
+
+    if(_tagAccordion.queryItem(category) == null) {
+      var tagGroup = AccordionItem(category, DivElement()..innerText = category, DivElement()..className = 'tags-menu__container', false, dataId: category);
+      _tagAccordion.appendItem(tagGroup);
+    }
+
+    var currentAccordion = _tagAccordion.queryItem(category);
+    currentAccordion.bodyElement.append(tag.renderElement);
+
+    // todo: potentially move this computation to onExpand
+    // recompute the grid
+    var wasHidden = !currentAccordion.isOpen;
+    if(wasHidden) {
+      currentAccordion.expand();
+    }
+    List<num> widths = currentAccordion.bodyElement.querySelectorAll('.tag__text').toList().map((e) => e.getBoundingClientRect().width).toList();
+    if(wasHidden) {
+      currentAccordion.collapse();
+    }
+    if(widths.isEmpty) {
+      widths.add(1);
+    }
+    num totalGridWidth = widths.fold(0, (previousValue, width) => previousValue + width);
+    num avgGridWidth = totalGridWidth / widths.length;
     num colSpacing = 10;
     num minColWidth = math.min(avgGridWidth + 2 * colSpacing, 138);
     num containerWidth = _tagsMenuWrapper.getBoundingClientRect().width;
     num columnWidth = containerWidth / (containerWidth / minColWidth).floor() - colSpacing;
-    _tagGroupsContainers[category].style.setProperty('grid-template-columns', 'repeat(auto-fill, ${columnWidth}px)');
+    currentAccordion.bodyElement.dataset['width'] = "${columnWidth}";
+    currentAccordion.bodyElement.style.setProperty('grid-template-columns', 'repeat(auto-fill, ${columnWidth}px)');
   }
 
   void removeMenuTag(FilterMenuTagView tag, String category) {
-    int index = _tagGroups[category].indexWhere((t) => t.renderElement.dataset["id"] == tag.renderElement.dataset["id"]);
-    _tagGroups[category][index].renderElement.remove();
-    _tagGroups[category].removeAt(index);
+    _tagAccordion.items.forEach((accordion) {
+      accordion.bodyElement.children.removeWhere((element) => element.dataset["id"] == tag.renderElement.dataset["id"]);
+    });
```

Comment:
This doesn't do the same thing that the older code does - it should be only the tag from the accordion with the id equal to `category` that gets removed, not from all accordion items.

Reply:
Fixed this!